### PR TITLE
MEF latest

### PR DESCRIPTION
**Pull request by rerowep**
* Implements a function and REST api to find the latest MEF with
  given entity type and pid.

### CODE REVIEW

File: rero_mef/agents/mef/api.py [hunk 1]
```diff
@@ -183,6 +183,33 @@ def create_deleted(cls, record, dbcommit=False, reindex=False):
...
 
+    @classmethod
+    def get_latest(cls, pid_type, pid):
+        """Get latest Mef record for pid_type and pid.
+
+        :param pid_type: pid type to use.
+        :param pid: pid to use..
+        :returns: latest record.
+        """
+        search = AgentMefSearch().filter({'term': {f'{pid_type}.pid': pid}})
```
Comment (zannkukai) [inline rero_mef/agents/mef/api.py:194]: Is moustache are necessary ? `AgentMefSearch().filter('term', f'{pid_type}.pid'=pid)` ?

File: rero_mef/agents/mef/api.py [hunk 2]
```diff
@@ -183,6 +183,33 @@ def create_deleted(cls, record, dbcommit=False, reindex=False):
...
+        if search.count() > 0:
+            data = next(search.scan()).to_dict()
+            new_pid = None
+            if relation_pid := data.get(pid_type, {}).get('relation_pid'):
```
Comment (zannkukai) [inline rero_mef/agents/mef/api.py:198]: COSMETIC : As you filter the query on `{pid_type}.pid`, you can be sure that this key exists. You could use simpler `data[pid_type].get('relation_pid')`. But your solution works well too ;-)

File: rero_mef/agents/mef/api.py [hunk 3]
```diff
@@ -183,6 +183,33 @@ def create_deleted(cls, record, dbcommit=False, reindex=False):
...
+                if relation_pid['type'] == 'redirect_to':
```
Comment (zannkukai) [inline rero_mef/agents/mef/api.py:199]: both if statement could be merge : 
```python
if (relation_pid := data[pid_type].get('relation_pid')) and relation_pid['type'] == 'redirect_to':
    new_pid = relation_pid['value']
```

File: rero_mef/agents/mef/api.py [hunk 4]
```diff
@@ -183,6 +183,33 @@ def create_deleted(cls, record, dbcommit=False, reindex=False):
...
+                    new_pid = relation_pid['value']
+            elif pid_type == 'idref':
+                # Find new pid from redirect_pid redirect_from
+                search = AgentMefSearch() \
+                    .filter('term', idref__relation_pid__value=pid)
+                if search.count() > 0:
+                    new_data = next(search.scan()).to_dict()
+                    new_pid = new_data.get('idref', {}).get('pid')
```
Comment (zannkukai) [inline rero_mef/agents/mef/api.py:207]: COSMETIC : could be `new_data['idref'].get('pid')`

File: rero_mef/agents/mef/views.py [hunk 5]
```diff
@@ -44,3 +44,17 @@ def redirect_item(pid):
...
+    'api_agents_mef_latest',
+    __name__,
+    url_prefix='/agents/mef'
+)
+
+
+@api_blueprint_latest.route('/latest/<pid_type>:<pid>')
+def get_latest(pid_type, pid):
+    """Get latest MEF for pid and type."""
+    from rero_mef.agents.mef.api import AgentMefRecord
```
Comment (zannkukai) [inline rero_mef/agents/mef/views.py:59]: Is this import can't be move to the top of the file ? I don't understand where a circular import could be occurred.

**Review by zannkukai — APPROVED** [review]

File: rero_mef/agents/mef/api.py (hunk 1 above)
Comment (rerowep) [reply rero_mef/agents/mef/api.py:194]: I think so.

File: rero_mef/agents/mef/api.py (hunk 3 above)
Comment (rerowep) [reply rero_mef/agents/mef/api.py:199]: Much to long the merge.

**Pull request closed without merging**